lang/py/avro/tether/tether_task.py

```diff
@@ -339,18 +341,15 @@ def configure(self,taskType,  inSchemaText,  outSchemaText):
       estr= traceback.format_exc()
       self.fail(estr)
 
-  def set_partitions(self,npartitions):
-
-    try:
-      self._partitions=npartitions
-    except Exception as e:
-      estr= traceback.format_exc()
-      self.fail(estr)
-
-  def get_partitions():
-    """ Return the number of map output partitions of this job."""
+  @property
+  def partitions(self):
+    """Return the number of map output partitions of this job."""
     return self._partitions
 
+  @partitions.setter
```

Comment:
Never seen this one before, looks nice! 

Ref: https://stackoverflow.com/questions/598077/why-does-foo-setter-in-python-not-work-for-me